hibernate-core/src/main/java/org/hibernate/dialect/lock/PessimisticReadSelectLockingStrategy.java

```diff
@@ -57,6 +57,7 @@ public void lock(Object id, Object version, Object object, int timeout, EventSou
 		final String sql = determineSql( timeout );
 		final SessionFactoryImplementor factory = session.getFactory();
 		try {
+			flushIfRequired( object, session );
```

Comment:
IMO the call to `flushIfRequired` should be moved to either `org.hibernate.loader.ast.internal.LoaderHelper#upgradeLock` or `org.hibernate.event.internal.DefaultLockEventListener#onLock` where we have the `EntityEntry` already available.

Reply:
changed it, WDYT @beikov ?